src/Sentry/Protocol/Envelope.cs

```diff
@@ -75,5 +79,24 @@ public static Envelope FromEvent(SentryEvent @event)
 
             return new Envelope(header, items);
         }
+
+        public static async Task<Envelope> DeserializeAsync(
+            Stream stream,
+            CancellationToken cancellationToken = default)
+        {
+            // Header
+            var header = await Json.DeserializeFromStreamAsync<Dictionary<string, object>>(stream, cancellationToken)
+                .ConfigureAwait(false);
+
+            // Items
+            var items = new List<EnvelopeItem>();
+            while (stream.ReadByte() == (byte)'\n')
```

Comment:
Consider there might be a trailing `\n` and no more items and that should be OK (as in don't throw)

Reply:
Reworked that entire part